tests/general/FilenameBearTest.py

```diff
@@ -63,8 +62,58 @@ def test_space_case(self):
         self.check_validity(self.uut, [''], filename='/a/Space Case')
 
     def test_ignore_upper(self):
+        self.section['file_naming_convention'] = 'auto'
         self.check_validity(self.uut, [''], filename='/LICENSE')
 
         self.section['ignore_uppercase_filenames'] = 'nope'
 
         self.check_validity(self.uut, [''], filename='/LICENSE', valid=False)
+
+    def test_default_naming_java(self):
+        self.section['file_naming_convention'] = 'auto'
+        self.check_validity(self.uut, [''], filename='/Home/xyz/x_y.java',
+                            valid=False)
+        self.check_validity(self.uut, [''], filename='/Home/xyz/x-y.java',
+                            valid=False)
+        self.check_validity(self.uut, [''], filename='/Home/xyz/x y.java',
+                            valid=False)
+        self.check_validity(self.uut, [''], filename='/Home/xyz/XY.java')
+
+    def test_default_naming_javascript(self):
+        self.section['file_naming_convention'] = 'auto'
+        self.check_validity(self.uut, [''], filename='/Home/xyz/x_y.js',
+                            valid=False)
+        self.check_validity(self.uut, [''], filename='/Home/xyz/x y.js',
+                            valid=False)
+        self.check_validity(self.uut, [''], filename='/Home/xyz/XY.js')
+        self.check_validity(self.uut, [''], filename='/Home/xyz/x-y.js')
+
+    def test_default_file_naming_python(self):
+        self.section['file_naming_convention'] = 'auto'
+        self.check_validity(self.uut, [''], filename='/Home/xyz/x_y.py')
+        self.check_validity(self.uut, [''], filename='XYZ/__init__.py')
+        self.check_validity(self.uut, [''], filename='/a/x y.py', valid=False)
+
+    def test_none_file_naming_convention_warning(self):
+        self.uut.execute('filename', [])
+
+        # Automatic bear messages we want to ignore.
+        self.uut.message_queue.get(timeout=1)
+
+        log = self.uut.message_queue.get(timeout=1)
+        self.assertEqual(log.message,
+                         'Please specify a file naming convention explicitly'
+                         ' or use "auto".')
+        self.check_validity(self.uut, [''], filename='/Home/xyz/x_y.py')
+
+    def test_auto_file_naming_convention_warning(self):
+        self.section['file_naming_convention'] = 'auto'
+        self.uut.execute('filename.xyz', [])
+
+        # Automatic bear messages we want to ignore.
+        self.uut.message_queue.get(timeout=1)
```

Comment:
do you have to specify a timeout? I would invoke that without timeout, our tests do run max 40secs before pytest aborts the test ;)